eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelTypes.scala

```diff
@@ -200,7 +203,20 @@ final case class LocalParams(nodeId: PublicKey,
                              isFunder: Boolean,
                              defaultFinalScriptPubKey: ByteVector,
                              globalFeatures: ByteVector,
-                             localFeatures: ByteVector)
+                             localFeatures: ByteVector) {
+
+  def fundingPubKey(keyManager: KeyManager) = keyManager.fundingPublicKey(fundingKeyPath)
+
+  def channelKeyPath(keyManager: KeyManager) = version match {
+    case 0 =>
+      // legacy mode:  we reuse the funding key path as our channel key path
+      fundingKeyPath
+    case 1 =>
+      // deterministic mode: use the funding pubkey to compute the channel key path
+      KeyManager.channelKeyPath(fundingPubKey(keyManager))
```

Comment:
Would it be possible to store the channel key path on a lazy val to avoid doing the `hash160` each time is used?